websites/D/Dumpert/presence.ts

```diff
@@ -0,0 +1,94 @@
+const presence = new Presence({
+  clientId: "840126038205923369"
+}),
+browsingStamp = Math.floor(Date.now() / 1000);
+let title: Element, title2: Element;
+
+presence.on("UpdateData", async () => {
+const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  },
+  page = window.location.pathname,
+  pageh = document.location.href;
+presenceData.startTimestamp = browsingStamp;
+
+function waitForElement(){
+  if(title2 != undefined){
+    if (title2.className.includes("paused")) {
+      delete presenceData.endTimestamp;
+      presenceData.smallImageKey = "pause";
+    } else if (title2.className.includes("playing")) {
+      const currentTime = presence.timestampFromFormat(
+        document.querySelector(
+          `#vjs_video_${title2.className.slice(40, 55).replace(/[^0-9.]/g, "")} > div.vjs-control-bar.progress-in-menu > div.vjs-current-time.vjs-time-control.vjs-control > span.vjs-current-time-display`
+        ).textContent
+      ),
+      
+      durationss = presence.timestampFromFormat(
+        document.querySelector(
+          `#vjs_video_${title2.className.slice(40, 55).replace(/[^0-9.]/g, "")} > div.vjs-control-bar.progress-in-menu > div.vjs-duration.vjs-time-control.vjs-control > span.vjs-duration-display`
+        ).textContent
+      ),
+      timestamps = presence.getTimestamps(currentTime, durationss);
+      presenceData.endTimestamp = timestamps[1];
+      presenceData.smallImageKey = "play";
+    }
+  }
+  else{
+      setTimeout(waitForElement, 3000);
+  }
+}
```

Comment:
Well what's the problem in this function. event `UpdateData` is already fired every i think 100ms if i am not wrong. so you literally don't need this.

Reply:
How should I do it then? If I don't do it like this the variable will return undefined.

Reply:
if undefined don't show anything?

Reply:
It loads after 3 seconds of the page being loaded.

Reply:
~~loop in loop isn't always the best~~